ts/torch_handler/base_handler.py

```diff
@@ -141,7 +152,10 @@ def preprocess(self, data):
         Returns:
             tensor: Returns the tensor data of the input
         """
-        return torch.as_tensor(data, device=self.device)
+        t = torch.as_tensor(data, device=self.device)
+        if ipex_enabled and t.dim() == 4:
+            t = t.to(memory_format=torch.channels_last)
```

Comment:
Suggest not do this here, reasons:
1. Input will be converted to channels last automatically inside the op as long as `model.to(memory_format=torch.channels_last)` (https://github.com/pytorch/serve/pull/1319/files#diff-d30e1f5ef9fd05e2ab9f652c116461632586cc132dde5d23ac4bc5cd19c799ccR85)
2. The channels last conversion only benefits CNN models while we can't tell if the model is CNN just by its input shapes. Moreover, CNN models can also take 3D or 5D inputs. Checking 4D sounds too specific to me.

Reply:
I definitely agree we can't assume CNN based on the input dimension being 4D (e.g., CNN for videos can take 5D inputs).